tests/integration/pilot/vm/README.md

```diff
@@ -0,0 +1,82 @@
+# Writing Tests for VMs
+
+Support for including VMs in the integration tests has been added to the framework.
+This document describes the integration and extension mechanisms to exercise VM related code.
+The primary goals are to test VM-related code & ensure it works on a range of supported OS types and versions.
+
+**Note: We currently use mock/simulated VMs for testing purposes. In the future, the testing might switch
+to utilize actual compute instances from different providers.**
+
+## Overview
+
+This section describes the scenarios in which one might want to add a VM test:
+1. Testing existing core Istio features such as traffic management, security, telemetry, etc. for **VMs**
+1. Supporting new OS images for VMs
+1. Testing onboarding tools (iptables, certs, istio-sidecar, etc.) and workflows (services, DNS, etc.) to enmesh a VM
+
+## Secenario 1: Testing VM-related Istio Code
+
+Most integration tests in Istio use the Echo application. To test connectivity, security and telemetry for a VM
+in the mesh, we deploy an instance of the Echo application as a VM resource. A VM Echo instance will simulate a VM,
+disabling kube-dns, Service Account mount, etc. For more information around VM onboarding,
+refer to this [doc](https://istio.io/latest/docs/examples/virtual-machines/single-network/).
+
+To deploy an echo instance as a VM
+1. Set the ports for the VMs. Note that due to a bug in WorkloadEntry,
+we need to set the ServicePort to be the same as the InstancePort
+1. Set `DeployAsVm` to be true in `echo.Config`
+1. Set `VMImage` to be any image available from `GetSupportedOSVersion` in [util.go](https://github.com/istio/istio/blob/master/tests/integration/pilot/vm/util.go).
+
+For example,
+
+```go
+ports := []echo.Port{
+   {
+       Name:     "http",
+       Protocol: protocol.HTTP,
+       // Due to a bug in WorkloadEntry, service port must equal target port for now
+       InstancePort: 8090,
+       ServicePort:  8090,
+   },
+}
+echo.Config{
+   Service:    "vm",
+   Namespace:  "virtual-machine",
+   Ports:      ports,
+   Pilot:      p,
+   DeployAsVM: true,
+   VMImage:    "app_sidecar_ubuntu_bionic"
+}
+```
+
+The default image referenced with `DefaultVMImage` should be used in most cases. If you want to use additional images
```

Comment:
What does 'most cases' refer to? Can you just clarify here the default image is used in all presubmit tests + provide additional information on when we would not?